crates/backend/src/lib.rs

```diff
@@ -19,42 +21,65 @@ use render::Renderer;
     let (sender, mut receiver) = unbounded_channel();
-    let server = Server::init(sender).await?;
+
+    let server = Server::init(sender.clone()).await?;
     info!("Backend: Server initialized");
 
     let (server_internal_channel, mut renderer) = Renderer::init(config)?;
     info!("Backend: Renderer initialized");
 
     let backend_thread = thread::spawn(move || renderer.run());
 
+    let mut scheduler = Scheduler::new();
+    info!("Backend: Scheduler initialized");
+
     loop {
         while let Ok(action) = receiver.try_recv() {
             match action {
                 Action::Show(notification) => {
+                    debug!(
+                        "Backend: Sent a request to renderer to show notification with id: {}",
+                        &notification.id
+                    );
                     server_internal_channel.send_to_renderer(
                         internal_messages::ServerMessage::ShowNotification(notification),
                     )?;
-                    debug!("Backend: Sended notification to render to show");
                 }
                 Action::Close(Some(id)) => {
                     server_internal_channel.send_to_renderer(
                         internal_messages::ServerMessage::CloseNotification { id },
                     )?;
-                    debug!("Backend: Sended notification with id {id} to render to close it");
+                    debug!(
+                        "Backend: Sent a request to renderer to close notification with id: {id}"
+                    );
                 }
-                Action::Close(None) => {
-                    warn!("Backend: Unsupported method 'Close'. Ignored");
+                Action::Schedule(notification) => {
+                    debug!(
+                        "Backend: Scheduled notification with id {} for time {}",
+                        &notification.id, &notification.time
+                    );
+                    scheduler.add(notification);
                 }
-                Action::ShowLast => {
-                    //TODO: make decision about this action. It may be very useless
-                    warn!("Backend: Unsupported method 'ShowLast'. Ignored");
+                Action::Close(None) => {
+                    warn!("Backend: Received 'Close' action without an id. Ignored");
                 }
                 Action::CloseAll => {
-                    warn!("Backend: Unsupported method 'CloseAll'. Ignored");
+                    warn!("Backend: Received unsupported 'CloseAll' action. Ignored");
                 }
             }
         }
 
+        scheduler
+            .pop_due_notifications()
+            .into_iter()
+            .for_each(|notification| {
+                debug!(
+                    "Backend: Notification with id {} due for delivery. Sending to renderer",
+                    &notification.id
+                );
+                sender.send(Action::Show(notification.data)).unwrap();
```

Comment:
We've talked about it and now I think that we should violate DRY principle because it will be cheaper, less complex and more readable. So instead of using sender, you're free to use `server_internal_channel` and directly send into renderer.

And don't forget about removing `.clone()` call above.